MetaMask/metamask-extension
# fix: update input validation

## **Description**

The trailing zeros automatically get removed when the user backspaces such that they exposes a zero, but a direct fix for this issue is blocked by the fact that the number input prevents trailing decimals (e.g., `1.`, `0.`)

This PR resolves both issues by programmatically updating the input field IFF there is an upstream update the current input is not equal to – upstream updates are needed for the max button to retain full functionality.



[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23788?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to the send flow and make the token primary (if it isn't already)
2. Type in a decimal with multiple trailing zeros followed by a non-zero number (e.g., 1.000001)
3. Backspace the last digit and ensure the trailing zeros remain
4. Continue to backspace to the decimal and ensure it isn't automatically removed

## **Screenshots/Recordings**



### **Before**

https://github.com/MetaMask/metamask-extension/assets/44588480/11678e8d-b0e2-4998-b14d-ea7f655e7365




### **After**



https://github.com/MetaMask/metamask-extension/assets/44588480/3de99210-5cc1-4213-924d-f6c7eeeaaa48


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
